org.eclipse.dawnsci.analysis.dataset/src/org/eclipse/dawnsci/analysis/dataset/slicer/SliceViewIterator.java

```diff
@@ -64,16 +64,12 @@ public SliceViewIterator(ILazyDataset lazyDataset, SliceND sampling, int... axes
 		
 		next = iterator.hasNext();
 		
-		List<SliceFromSeriesMetadata> sl;
-		try {
-			sl = lazyDataset.getMetadata(SliceFromSeriesMetadata.class);
-			if (sl != null && !sl.isEmpty() && sl.get(0) != null) {
-				SliceFromSeriesMetadata ss = sl.get(0);
+
+			SliceFromSeriesMetadata ss = lazyDataset.getFirstMetadata(SliceFromSeriesMetadata.class);
+			if (ss != null) {
 				if (ss.getSourceInfo() != null) source = ss.getSourceInfo();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1066)